HW3/task1/eventEmitter.js

```diff
@@ -0,0 +1,61 @@
+class EventEmitter {
+  listeners = {}; // key-value pair
+
+  addListener(eventName, fn) {
+    this.listeners[eventName] = this.listeners[eventName] || [];
+    this.listeners[eventName].push(fn);
+    return this;
+  }
+
+  on(eventName, fn) {
+    return this.addListener(eventName, fn);
+  }
+
+  removeListener(eventName, fn) {
+    let lstnr = this.listeners[eventName];
+    if (lstnr) {
+      for (let i = lstnr.length - 1; i >= 0; i--) {
+        if (lstnr[i] === fn) {
+          lstnr.splice(i, 1);
+        }
+      }
+    }
+    return this;
+  }
+
+  off(eventName, fn) {
+    return this.removeListener(eventName, fn);
+  }
+
+  once(eventName, fn) {
```

Comment:
How can we remove listener for `once` events?

Reply:
Updated